src/main/java/org/springframework/hateoas/GenericAffordanceModel.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.hateoas;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+import java.util.List;
+
+import org.springframework.core.ResolvableType;
+import org.springframework.http.HttpMethod;
+
+/**
+ * Collection of attributes needed to render any form of hypermedia.
+ * 
+ * @author Greg Turnquist
+ */
+@EqualsAndHashCode
+@AllArgsConstructor
+public class GenericAffordanceModel {
```

Comment:
Probably because it used to be tiny interface. I created what would be an abstract base class with nothing abstract and realized I didn’t need the interface anymore.

It makes sense to rename it as AffordanceModel and not give the impression there is something further up the type hierarchy. 

Additionally, since no one is supposed to use the class directly, it should probably me marked abstract even if there are no abstract methods.